src/databricks/labs/ucx/azure/resources.py

```diff
@@ -140,16 +181,64 @@ def post(self, path: str, body: dict | None = None):
             return self.api_client.do("POST", path, query, headers, body)
         return self.api_client.do("POST", path, query, headers)
 
-    def delete(self, path: str):
+    def delete(self, path: str, api_version: str | None = None):
         # this method is added only to be used in int test to delete the application once tests pass
         headers = {"Content-Type": "application/json"}
         query: dict[str, str] = {}
+        if api_version is not None:
+            query = {"api-version": api_version}
         return self.api_client.do("DELETE", path, query, headers)
 
     def token(self):
         return self._token_source.token()
 
 
+class AccessConnectorClient:
+
+    def __init__(self, azure_mgmt: AzureAPIClient) -> None:
+        self._api_version = "2023-05-01"
+        self._azure_mgmt = azure_mgmt
+
+    def list(self, subscription_id: str) -> list[AccessConnector]:
+        """List all access connector within subscription
+        
+        Docs:
+            https://learn.microsoft.com/en-us/rest/api/databricks/access-connectors/list-by-subscription?view=rest-databricks-2023-05-01&tabs=HTTP    
+        """
+        url = f"/subscriptions/{subscription_id}/providers/Microsoft.Databricks/accessConnectors"
+        response = self._azure_mgmt.get(url, self._api_version)
+
+        access_connectors = []
+        for access_connector_raw in response["value"]:
+            access_connector = AccessConnector(
+                id=access_connector_raw["id"],
+                type=access_connector_raw["type"],
+                location=access_connector_raw["location"],
+            )
+            access_connectors.append(access_connector)
+        return access_connectors
+
+    def create(self, access_connector: AccessConnector):
```

Comment:
```suggestion
    def create(self, subscription, resource_group, name, location, managed_identity, properties, tags):
```
Don't pass in single POJO, pass in arguments